sns.py

```diff
@@ -0,0 +1,58 @@
+def positive1(self, sqs_arn: str, topic_arn: str) -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    # create SNS topic, connect it to the Lambda, publish test message
+    sns_client.subscribe(TopicArn=topic_arn, Protocol='sqs',
+                         Endpoint=sqs_arn)
+
+def positive2(self, sqs_arn: str, topic_arn: str) -> None:
+    session = boto3.Session()
+    sns_client = session.client('sns')
+    response = sns_client.subscribe(
+        TopicArn=topic_arn,
+        Protocol='sqs',
+        Endpoint=sqs_arn,
+        ReturnSubscriptionArn=True)
+    return response
+
+def positive3() -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    return sns_client.create_topic(Name=TEST_TOPIC_NAME)
+
+def negative1() -> None:
+    session = boto3.Session()
+    sns_client = session.client('sns')
+    topic_info = sns_client.create_topic(Name=TEST_TOPIC_NAME)
+    sns_client.subscribe(TopicArn=topic_info['TopicArn'], Protocol='sqs',
+                         Endpoint=aws_stack.sqs_queue_arn(TEST_QUEUE_NAME_FOR_SNS))
+    test_value = short_uid()
+    sns_client.publish(TopicArn=topic_info['TopicArn'], Message='test message for SQS',
+                       MessageAttributes={'attr1': {'DataType': 'String', 'StringValue': test_value}})
+
+def negative2() -> None:
+    session = botocore.session.get_session()
+    sns_client = session.create_client('sns', 'us-west-2')
+    # create SNS topic, connect it to the Lambda, publish test message
+    num_events_sns = 3
+    response = sns_client.create_topic(Name='TEST_TOPIC_NAME')
+    sns_client.subscribe(TopicArn=response['TopicArn'], Protocol='lambda',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The Amazon SNS subscribe operation by default returns either the subscription ARN (if the subscribed endpoint is managed by AWS and it belongs to the same account as the topic) or the phrase: `PENDING CONFIRMATION`. If you want to always return the subscription ARN, set the `ReturnSubscriptionArn` argument to `True`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#SNS.Client.subscribe)